polkadot/node/network/approval-distribution/src/lib.rs

```diff
@@ -1319,6 +1322,33 @@ impl State {
 		self.enable_aggression(network_sender, Resend::No, metrics).await;
 	}
 
+	// When finality is lagging as a last resort nodes start sending the messages they have
+	// multiples times. This means it is safe to accept duplicate messages without punishing the
+	// peer and reduce the reputation and can end up banning the Peer, which in turn will create
+	// more no-shows.
+	fn accept_duplicates_from_validators(
```

Comment:
It would safer to accept only a bounded amount of duplicates over time.

Reply:
I think that would complicate even more this solution, given that this is the last resolution and I would want to catch this release with this improvement I'll add a ticket to further refine it in the future.